pkg/builder/local_build_executor_test.go

```diff
@@ -583,15 +583,23 @@ func TestLocalBuildExecutorWithWorkingDirectorySuccess(t *testing.T) {
 	// Creation of build/objects
 	rootDirectory := mock.NewMockDirectory(ctrl)
 	rootDirectory.EXPECT().Mkdir("build", os.FileMode(0777)).Return(nil)
+	rootDirectory.EXPECT().Mkdir("build", os.FileMode(0777)).Return(nil)
 	buildDirectory := mock.NewMockDirectory(ctrl)
 	rootDirectory.EXPECT().Enter("build").Return(buildDirectory, nil)
+	rootDirectory.EXPECT().Enter("build").Return(buildDirectory, nil)
```

Comment:
Could you please keep this ordered a bit closer to what happens in practice? Also: maybe it's a bit more accurate to use two separate mocks for `buildDirectory` instead of one. In practice, it won't really happen that `Enter()` returns the same handle for both `Enter()` calls.

Reply:
As part of making this test a bit more realistic I'll probably move away from needing to call `Enter("build")` twice so that should handle this issue.